pkg/uidmap/uidmap.go

```diff
@@ -0,0 +1,136 @@
+package uidmap
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/docker/libcontainer/user"
+)
+
+func xlateOneFile(path string, finfo os.FileInfo, containerRoot uint32, inverse bool) error {
+	uid := uint32(finfo.Sys().(*syscall.Stat_t).Uid)
+	gid := uint32(finfo.Sys().(*syscall.Stat_t).Gid)
+	mode := finfo.Mode()
+
+	if (uid == containerRoot || gid == containerRoot) && !inverse {
+		fmt.Errorf("Warning: UID of an existing file (%s) is docker-root")
+	}
+	if ((uid == 0 || gid == 0) && !inverse) || ((uid == containerRoot || gid == containerRoot) && inverse) {
+		newUid := uid
+		newGid := gid
+		if uid == 0 && !inverse {
+			newUid = containerRoot
+		}
+		if gid == 0 && !inverse {
+			newGid = containerRoot
+		}
+		if uid == containerRoot && inverse {
+			newUid = 0
+		}
+		if gid == containerRoot && inverse {
+			newGid = 0
+		}
+		if err := os.Lchown(path, int(newUid), int(newGid)); err != nil {
+			return fmt.Errorf("Cannot chown %s: %s", path, err)
+		}
+		if err := os.Chmod(path, mode); err != nil {
+			return fmt.Errorf("Cannot chmod %s: %s", path, err)
+		}
+	}
+
+	return nil
+}
+
+func xlateUidsRecursive(base string, containerRoot uint32, inverse bool) error {
+	f, err := os.Open(base)
+	if err != nil {
+		return err
+	}
+
+	list, err := f.Readdir(-1)
+	f.Close()
+	if err != nil {
+		return err
+	}
+
+	for _, finfo := range list {
+		path := filepath.Join(base, finfo.Name())
+		if finfo.IsDir() {
+			xlateUidsRecursive(path, containerRoot, inverse)
```

Comment:
It's an artifact of initial code that would keep going even if it stumbles on one or two (special) files across all files in the file system.  Will fix it.